### ISSUE TITLE
Imports column all zeroes in two-region Use table

### ISSUE THREAD

**Issue by WesIngwersen** [USEPA/stateior#14]

**Comment by MoLi7** [comment]
This issue was reported by Andre and Ming. I cannot find in our exchanged emails or my notes about what two-region Use table they referred to, but I think they referred to the (total) Use table rather than the domestic Use table, because import column is by definition 0 in the domestic Use table.

Having the two-region (total) Use table as the object, I suspected the root cause for zeros in import column (F050) being [allocation ratios are 0 for regionalizing US international imports](https://github.com/USEPA/stateior/blob/dev/R/BuildModel.R#L430-L444).

**Comment by MoLi7** [comment]
Before finding a way to modify the ratio matrix mentioned above, I want to take a step back to fully understand, and hopefully explain, why and how international import needs to be presented in the two-region Use table.

Indeed, we need to allocate national import to states as we build state Use models, which has been done and [validated by this check](https://github.com/USEPA/stateior/blob/dev/inst/doc/ValidateModel.Rmd#L206-L237). But for two-region IO models, in fact [import is excluded from sectors that are considered tradable between SoI and RoUS](https://github.com/USEPA/stateior/blob/dev/R/BuildModel.R#L254-L258) from the very beginning. To me, this implies we assume international import cannot be traded from SoI to RoUS or the other way around, which means US import matrix should only be allocated to SoI-to-SoI and RoUS-to-RoUS Use tables to reflect such non-tradable essence.

If this makes sense, US import matrix should only be broken into an SoI import matrix and an RoUS import matrix using the proportions of international imports (F050) in SoI and in RoUS, respectively like so.
```
SoI import matrix = SoI_F050/US_F050 * US import matrix
RoUS import matrix = RoUS_F050/US_F050 * US import matrix
```

Then the SoI and RoUS import matrices will be added back to SoI-to-SoI Use and RoUS-to-RoUS Use tables, while leaving the SoI-to-RoUS and RoUS-to-SoI tables unchanged, to eventually form the two-region (total) Use table.

**Comment by WesIngwersen** [comment]
@MoLi7 
Let me be clear that I understand this approach will not yield two import matrices that can be summed to equal the US import matrix as provided by BEA.  That's because we already treated F050 differently in the state models.  And because we know that the National Use table F050 table and the BEA Import matrix have differences themselves in the margin sectors. 

So this is acceptable and expected.

**Comment by MoLi7** [comment]
> Let me be clear that I understand this approach will not yield two import matrices that can be summed to equal the US import matrix as provided by BEA. 

My understanding is, the goal actually is to create two import matrices that sum up to the US import matrix. Because the import matrix itself represents the biggest difference between a Use and a DomesticUse (a smaller difference is about the margin sectors). Though the challenge does not lie in "state F050 were handled differently", instead the challenge is the ratio for splitting, i.e. `SoI_F050/US_F050` and `RoUS_F050/US_F050`, because there are positive and negative values and zeros in the F050 vectors, which causes the ratio derivation not easily achievable.

> That's because we already treated F050 differently in the state models.

Well, we estimated F050 in each state by subtracting **Use's commodity output less F050** from **Make's commodity output**. This is different from "allocating US F050 to states", but because we've confirmed state Use and Make sum to US Use and Make, respectively, sum of state F050 is found close to US F050 with gaps between +/- $10 million.

> And because we know that the National Use table F050 table and the BEA Import matrix have differences themselves in the margin sectors.

This difference is represented by F051 which has been added in SoI_Use and can be easily generated by a function for US_Use, so F051 for RoUS_Use can created via a simple subtraction. Then in the two-region Use table, F051 in SoI_Use can be carried over to SoI-to-SoI Use, while F051 in RoUS_Use can be carried over to RoUS-to-RoUS Use.
